src/util/system.cpp

```diff
@@ -603,7 +603,21 @@ int64_t ArgsManager::GetIntArg(const std::string& strArg, int64_t nDefault) cons
 bool ArgsManager::GetBoolArg(const std::string& strArg, bool fDefault) const
 {
     const util::SettingsValue value = GetSetting(strArg);
-    return value.isNull() ? fDefault : value.isBool() ? value.get_bool() : InterpretBool(value.get_str());
+    switch (value.getType()) {
+        case UniValue::VNULL:
+            return fDefault;
+        case UniValue::VBOOL:
+            return value.get_bool();
+        case UniValue::VOBJ:
+        case UniValue::VARR:
+            // Throws an exception
+            value.get_str();
+            assert(false);
+        case UniValue::VSTR:
+        case UniValue::VNUM:
+            return InterpretBool(value.getValStr());
+    }
+    assert(false);
```

Comment:
In commit "Bugfix: util: Correctly handle Number value types in GetArg/GetBoolArg" (33722279495ebf566436a274765068217b2d3543)

This is pretty verbose and a little hard to follow. Could consider stealing Marco's style from https://github.com/bitcoin/bitcoin/pull/24498#discussion_r822889392 and writing something like:

```c++
return value.isNull() ? fDefault :
       value.isBool() ? value.get_bool() :
       value.isNum()  ? InterpretBool(value.getValStr()) :
                        InterpretBool(value.get_str());
```

Reply:
Huh? I changed it to a switch because it's very readable and the repeated trinary operator is hard to follow...